### ISSUE TITLE
Enhancement for ExecAction in Pod Lifecycle

### ISSUE THREAD

**Issue by quanweiZhou** [kubernetes/kubernetes#128250]
### What would you like to be added?

Currently the the ExecAction in Pod Lifecycle does not support timeout and the hardcode when run in the container. Suggest adding a timeout param fo ExecAction.

* https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json
```
"io.k8s.api.core.v1.ExecAction": {
      "description": "ExecAction describes a \"run in container\" action.",
      "properties": {
        "command": {
          "description": "Command is the command line to execute inside the container, the working directory for the command  is root ('/') in the container's filesystem. The command is simply exec'd, it is not run inside a shell, so traditional shell instructions ('|', etc) won't work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy.",
          "items": {
            "type": "string"
          },
          "type": "array",
          "x-kubernetes-list-type": "atomic"
        }
      },
      "type": "object"
    }
```
* https://github.com/kubernetes/kubernetes/blob/9d967ff97332a024b8ae5ba89c83c239474f42fd/pkg/kubelet/lifecycle/handlers.go#L72-L80

### Why is this needed?

If PostStart uses ExecAction and exec hangs during execution, the container call will hang. If the timeout is supported, users can force-stop the container based on the timeout.

### Labels:
/sig sig-api-machinery
/wg wg-api-expression

**Comment by saschagrunert** [comment]
I think this makes sense generally.

**Comment by BenTheElder** [comment]
> Enhancement for ExecAction in Pod Lifecycle

The process for enhancements is in:
https://github.com/kubernetes/enhancements

/remove-sig api-machinery
/remove-wg api-expression
/sig node

**Comment by googs1025** [comment]
/assign 
@BenTheElder 
I am willing to give it a try. Do I need to file an enhancement issue in https://github.com/kubernetes/enhancements first?

**Comment by carlory** [comment]
>  Do I need to file an enhancement issue in https://github.com/kubernetes/enhancements first?

@googs1025 Yes it is, creating a new issue in the enhancement repo to get the kep number.

**Comment by sreeram-venkitesh** [comment]
@googs1025 Yes, now that we're past enhancements freeze, you can try to get this in alpha for v1.33. You can join the SIG Node meetings on Tuesdays if you have any questions.

**Comment by thockin** [comment]
Without looking too deeply, it seems reasonable, but it will need a KEP.

**Comment by googs1025** [comment]
Thanks for the reply @thockin . I will propose a KEP for discussion soon.

**Comment by sreeram-venkitesh** [comment]
@googs1025 Are you planning to work on this? I see that the KEP you had created for this https://github.com/kubernetes/enhancements/issues/4929 has been closed as stale (I've reopened it).

Now that we're past PRR freeze for v1.37, we can try to aim for v1.38. I can help out too if you'd like.

**Comment by googs1025** [comment]
/remove-lifecycle stale

Thanks for reminder, I will address it in v1.38.
